### PR TITLE
Add Custom view transition capabilites

### PR DESCRIPTION
**Pull request by Que-tin**
[add] Custom view transition capabilites
[add] Custom view transitions example
[add] Reload & back / forward handling
[fix] Slide navigation out of document scope

### CODE REVIEW

File: src/styles/transitions.css [hunk 1]
```diff
@@ -1,65 +1,67 @@
...
-}
+	&[data-view-transition='element'] {
+		view-transition-name: none;
 
-.disable-element-transitions :is(h1, h2, pre, p, a, picture, aside) {
-  view-transition-name: none !important;
-}
+		& .disable-element-transitions :is(h1, h2, pre, p, a, picture, aside) {
+			view-transition-name: none !important;
+		}
```
Comment (argyleink) [inline src/styles/transitions.css]: we can get rid of this as it's now replaced with the data attribute strategy

**Review by argyleink — CHANGES_REQUESTED** [review]
all in all looks great. simplifies a feature and adds directionality awareness which is neat.

i think the last part might be to document it a bit? wanna add a tiny readme entry and then we'll merge this in?

**Comment by Que-tin** [comment]
Currently writing some documentation, actually coincidental didn't even see the comment.

**Comment by Que-tin** [comment]
@argyleink README.md changes ready for review.

**Review by argyleink — APPROVED** [review]

**Pull request merged**